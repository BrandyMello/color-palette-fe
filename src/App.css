@import url('https://fonts.googleapis.com/css?family=Modak|Ubuntu&display=swap');

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

body {
  background-color: #262629;
  color: white;
  /* font-family: 'Ubuntu',
    sans-serif; */
}