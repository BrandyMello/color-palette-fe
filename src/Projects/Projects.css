td {
  border: 1px solid black;
  border-radius: 8px;
  width: 300px;
  height:300px;
}

table {
  margin: 2%;
}

.project-name-h2 {
  font-size: 40px;
  text-align: center;
}

.palette-name {
  font-size: 30px;
}

.delete-palette {
  background-color: #808080;
  color: white;
  border: 1px solid black;
  border-radius: 5px;
  font-size: 14px;
  height: 40px;
  width: 100px;
}