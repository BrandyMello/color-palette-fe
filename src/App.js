import React, { Component } from "react";
import "./App.css";
import {
  getProjects,
  getAllPalettes,
  deleteProject,
  deletePalette,
  addNewProject,
  addNewPalette,
  updateProject,
  updatePalette
} from "./apiCalls/apiCalls";
import SavedProjectsNav from "./SavedProjectsNav/SavedProjectsNav";
import Projects from "./Projects/Projects";
import { Route, NavLink } from "react-router-dom";
import Nav from "./Nav/Nav";
import RandomPalette from "./RandomPalette/RandomPalette";
import SaveProjects from "./SaveProjects/SaveProjects";
import SavePalette from "./SavePalette/SavePalette";

class App extends Component {
  constructor() {
    super();
    this.state = {
      projects: [],
      projectName: "",
      palettes: [],
      colors: [
        { color_1: this.generateRandomHex() },
        { color_2: this.generateRandomHex() },
        { color_3: this.generateRandomHex() },
        { color_4: this.generateRandomHex() },
        { color_5: this.generateRandomHex() }
      ],
      message: ""
    };
  }

  async componentDidMount() {
    try {
      const projects = await getProjects();
      const palettes = await getAllPalettes();
      this.setState({ projects: projects });
      this.setState({ palettes: palettes });
    } catch ({ message }) {
      this.setState({ error: message });
    }
  }

  generateRandomHex = () => {
    const characters = "0123456789ABCDEF";
    let hex = "";
    for (let i = 0; i < 6; i++) {
      hex += characters[Math.floor(Math.random() * 16)];
    }
    return `#${hex}`;
  };

  generateRandomPalette = () => {
    this.setState({
      colors: [
        { color_1: this.generateRandomHex() },
        { color_2: this.generateRandomHex() },
        { color_3: this.generateRandomHex() },
        { color_4: this.generateRandomHex() },
        { color_5: this.generateRandomHex() }
      ]
    });
  };

  saveNewProject = async (name, e) => {
    e.preventDefault();
    try {
      await addNewProject(name);
      const projects = await getProjects();
      this.setState({ projects });
    } catch (error) {
      this.setState({ error });
    }
  };

  saveNewPaletteToProject = async paletteInfo => {
    try {
      await addNewPalette(paletteInfo);
      const palette = await getAllPalettes();
      this.setState({ palettes: { ...palette } });
    } catch (error) {
      this.setState({ error });
    }
  };

  handlePaletteNameChange = (e, oldPalette) => {
    const newPalette = { ...oldPalette, name: e.target.innerText };
    updatePalette(newPalette);
  };

  handleProjectNameChange = (e, props) => {
    const newProjectName = e.target.innerText;
    const keepPalettes = this.state.palettes.filter(
      palette => palette.projectId === props.id
    );
    keepPalettes.forEach(keptPalette =>
      updatePalette({ ...keptPalette, projectName: newProjectName })
    );
    const newProject = { id: props.id, name: newProjectName };
    updateProject(newProject);
  };

  deleteSpecificProject = async id => {
    try {
      await deleteProject(id);
      const projects = await getProjects();
      this.setState({ projects });
    } catch (error) {
      this.setState({ error });
    }
  };

  deleteSpecificPalette = async id => {
    try {
      await deletePalette(id);
      const palettes = await getAllPalettes();
      this.setState({ palettes });
    } catch (error) {
      this.setState({ error });
    }
  };

  render() {
    const { projects } = this.state;
    return (
      <div>
        <NavLink to="/">
          <button className="homeButton">HOME</button>
        </NavLink>
        <Nav />
        <main>
          <Route
            exact
            path="/"
            render={() => (
              <div>
                <SavedProjectsNav projects={this.state.projects} />
                <RandomPalette
                  colors={this.state.colors}
                  generateRandomPalette={this.generateRandomPalette}
                />
                <SaveProjects
                  projects={this.state.projects}
                  saveNewProject={this.saveNewProject}
                />
                <SavePalette
                  projects={this.state.projects}
                  colors={this.state.colors}
                  saveNewPaletteToProject={this.saveNewPaletteToProject}
                />
              </div>
            )}
          />
          <Route
            path="/projects/:id"
            render={({ match }) => {
              const { id } = match.params;
              const foundProject = projects.find(
                project => project.id === parseInt(id)
              );
              return (
                <div>
                  <Projects
                    {...foundProject}
                    palettes={this.state.palettes}
                    updateProjectName={this.updateProjectName}
                    deleteSpecificProject={this.deleteSpecificProject}
                    deleteSpecificPalette={this.deleteSpecificPalette}
                    handlePaletteNameChange={this.handlePaletteNameChange}
                    handleProjectNameChange={this.handleProjectNameChange}
                  />
                </div>
              );
            }}
          />
        </main>
      </div>
    );
  }
}

export default App;
