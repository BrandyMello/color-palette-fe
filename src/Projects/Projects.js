import React from 'react';
import './Projects.css';

const Projects = (props) => {
  console.log('props', props)
  let projectPalettes = props.palettes.filter(palette => {
    if(palette.projectName === props.name) {
      return palette
    }
  });
  let paletteRow = projectPalettes.map((projPalette, index) => {
   return (
     <>
     <tr>
       <th>{projPalette.name}</th>
    {/* <button type="submit" onclick={props.deleteSpecificProject(props.id)}>Delete Project</button> */}
     </tr>
   <tr key={projPalette.id}>
     <td key={index} style={{ backgroundColor: projPalette.colorOne }}>lock</td>
     <td key={index} style={{ backgroundColor: projPalette.colorTwo }}>lock</td>
     <td key={index} style={{ backgroundColor: projPalette.colorThree }}>lock</td>
     <td key={index} style={{ backgroundColor: projPalette.colorFour }}>lock</td>
     <td key={index} style={{ backgroundColor: projPalette.colorFive }}>lock</td>
     <button type="submit" onClick={() => props.deleteSpecificPalette(props.palettes[index].id)}>Delete Palette</button>
    </tr>
    </>
   )
  })
  return (
    <div>
      <h2 contentEditable={true}>{props.name}</h2>
      <table>
        {paletteRow}
      </table>
    </div>
  )
}

export default Projects