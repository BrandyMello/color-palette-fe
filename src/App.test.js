import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Projects from './Projects/Projects';
import App from './App';

describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Routes', () => {
  it('should route to People', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/projects/:id']}>
        <App />
      </MemoryRouter>
    )

    expect(wrapper.find(Projects)).toHaveLength(1)
  });