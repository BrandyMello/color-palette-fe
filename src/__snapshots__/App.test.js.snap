// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div>
  <NavLink
    to="/"
  >
    <button
      className="homeButton"
    >
      HOME
    </button>
  </NavLink>
  <Nav />
  <main>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      path="/projects/:id"
      render={[Function]}
    />
  </main>
</div>
`;
